### PR TITLE
Fixes for fast register

### PR DESCRIPTION
**Pull request by katrogan**

### CODE REVIEW

File: flytekit/annotated/base_task.py [hunk 1]
```diff
@@ -412,7 +413,7 @@ def get_registerable_entity(self) -> SdkTask:
 
     def get_fast_registerable_entity(self) -> SdkTask:
         entity = self.get_registerable_entity()
-        if entity.container is None:
+        if entity.container is None or entity.type == SdkTaskType.RAW_CONTAINER_TASK:
```
Comment (kumare3) [inline flytekit/annotated/base_task.py:416]: Why are we using this constant? can we use from Annotated?

Comment (katrogan) [reply flytekit/annotated/base_task.py]: good point! done

**Comment by katrogan** [comment]
ping @wild-endeavor @kumare3

File: flytekit/annotated/constants.py [hunk 2]
```diff
@@ -0,0 +1 @@
+RAW_CONTAINER_TASK_TYPE = "raw-container"
```
Comment (wild-endeavor) [inline flytekit/annotated/constants.py:1]: don't we have this already somewhere else?  can we use that one @kumare3

Comment (katrogan) [reply flytekit/annotated/constants.py:1]: it was defined inline in container_task.py

Comment (wild-endeavor) [reply flytekit/annotated/constants.py:1]: it's defined in `flytekit/common/constants.py:25`

Comment (wild-endeavor) [reply flytekit/annotated/constants.py:1]: okay apparently this is just a console thing, ignore my last comment.

Comment (kumare3) [reply flytekit/annotated/constants.py:1]: can we not define a constant again :(, it is part of the task

**Pull request merged**